test-support/src/main/java/org/testcontainers/testsupport/FlakyTestJUnit4RetryRule.java

```diff
@@ -0,0 +1,93 @@
+package org.testcontainers.testsupport;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.MultipleFailureException;
+import org.junit.runners.model.Statement;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * <p>
+ *     JUnit 4 @Rule that implements retry for flaky tests (tests that suffer from sporadic random failures).
+ * </p>
+ * <p>
+ *     This rule should be used in conjunction with the @{@link Flaky} annotation. When this Rule is applied to a test
+ *     class, any test method with this annotation will be invoked up to 3 times or until it succeeds.
+ * </p>
+ * <p>
+ *     Tests should <em>not</em> be marked @{@link Flaky} for a long period of time. Every usage should be
+ *     accompanied by a GitHub issue URL, and should be subject to review at a suitable point in the (near) future.
+ *     Should the review date pass without the test's instability being fixed, the retry behaviour will cease to have an
+ *     effect and the test will be allowed to sporadically fail again.
+ * </p>
+ */
+@Slf4j
+public class FlakyTestJUnit4RetryRule implements TestRule {
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+
+        final Flaky annotation = description.getAnnotation(Flaky.class);
+
+        if (annotation != null) {
+            if (annotation.githubIssueUrl().trim().length() == 0) {
+                throw new IllegalArgumentException("A GitHub issue URL must be set for usages of the @Flaky annotation");
+            }
+
+            final int maxTries = annotation.maxTries();
```

Comment:
Since we are validating the input arguments, does it make sense to also check `maxTries > 1`?

Reply:
Good idea, will do